src/widgets/modalChildren/PatientEditModal.js

```diff
@@ -9,27 +9,36 @@ import { FlexRow } from '../FlexRow';
 import { JSONForm } from '../JSONForm/JSONForm';
 import { NameNoteActions } from '../../actions/Entities/NameNoteActions';
 import { selectNameNoteIsValid, selectCreatingNameNote } from '../../selectors/Entities/nameNote';
+import { selectSortedPatientHistory } from '../../selectors/patient';
 import { selectCompletedForm, selectCanSaveForm } from '../../selectors/form';
 import { PatientActions } from '../../actions/PatientActions';
 import globalStyles, { SUSSOL_ORANGE } from '../../globalStyles';
-import { generalStrings, modalStrings } from '../../localization/index';
+import { generalStrings, modalStrings, navStrings } from '../../localization/index';
+import { PaperModalContainer } from '../PaperModal/PaperModalContainer';
+import { PaperConfirmModal } from '../PaperModal/PaperConfirmModal';
+import { useToggle } from '../../hooks/index';
 
 export const PatientEditModalComponent = ({
   isDisabled,
   onSaveForm,
+  onDeleteForm,
   onCancel,
   inputConfig,
   surveySchema,
   surveyForm,
   onUpdateForm,
   nameNoteIsValid,
   canSaveForm,
+  hasVaccineEventsForm,
 }) => {
   let canSave = canSaveForm;
+  const hasVaccineEvents = hasVaccineEventsForm;
```

Comment:
I think that instead of restricting to delete patients if there are vaccination events we should allow to delete of patient AND delete all the vaccination events if they were entered on the current tablet:
![Screen Shot 2022-07-22 at 2 15 02 PM](https://user-images.githubusercontent.com/41636693/180348216-e2343e50-89d0-4655-a8d5-26cc3e2db9f1.png)

However, if there are vaccination events entered on a different tablet, the user should be notified about it when clicking "Delete". I think making "Delete" enterable is actually better (and just show an alert saying that this patient cannot be deleted because there are vaccination events entered on a different tablet). Currently, it can be sometimes confusing for the user why they cannot delete the patient as there are no alerts.

Reply:
@chastichno I was also thinking of showing an alert instead of disabling delete button.
Onto the next point, since we already have individual vaccine event deletion, and it would be less likely for the user to group delete the whole set if we just prevent deletion until all vaccine events associated to the patient has been deleted. Also, since it's the same functionality as we have in mSupply desktop transaction deletion. Also it is easier to do it this way.

Let's keep it this way, I will add the popup notification saying you have to delete vaccine events first. We can update if client complains.

Reply:
Yeah, I agree it is actually easier not to delete the vaccination events for the users. But a pop-up window specifying the reason why they cannot delete a patient would be nice.

Reply:
Resolved with [983e600](https://github.com/openmsupply/mobile/pull/4812/commits/983e6006e7359a693d190312ab2f8a4e7777bf3e)